### DISCUSSION TITLE
Customising package installations via CMake

### DISCUSSION THREAD

**Discussion by ghost** [microsoft/vcpkg#18099] [Q&A] [answered]
My project depends on both `assimp` and `minizip-ng`, but the two conflict at linkage due to each defining the same bzip error callback. The fix isn't very difficult; instead of install `assimp` using vcpkg, I can submodule it, disable its Irrlicht importers using CMake variables and preprocessor definitions, and build it as such.

If I wanted to do the same thing through vcpkg with a workflow that would be easy for newly-arrived contributors to my project -- preferably entirely with CMake commands and/or a bat/shell script -- how would I do so?

**Comment by ras0219-msft** [comment, marked as answer]
You can use overlay ports for this -- your project can use a private version of assimp which has the customizations you need
